provider/core/src/main/java/org/arquillian/pact/provider/core/deployment/EnvironmentUrlResourceProvider.java

```diff
@@ -0,0 +1,61 @@
+package org.arquillian.pact.provider.core.deployment;
+
+
+import org.arquillian.pact.provider.api.deployment.Environment;
+import org.arquillian.pact.provider.core.PactProviderConfiguration;
+import org.jboss.arquillian.container.test.impl.enricher.resource.URLResourceProvider;
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.test.api.ArquillianResource;
+
+import java.lang.annotation.Annotation;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+
+public class EnvironmentUrlResourceProvider extends URLResourceProvider {
+
+    private static final int NOT_FOUND = -1;
+
+    @Inject
+    Instance<PactProviderConfiguration> pactProviderConfigurationInstance;
+
+    @Override
+    public Object doLookup(ArquillianResource resource, Annotation... qualifiers) {
+        if (DeploymentEnabler.shouldEnableDeployment(pactProviderConfigurationInstance.get())) {
+            return super.doLookup(resource, qualifiers);
+        } else {
+            int position = NOT_FOUND;
+            if ((position = getEnvironmentAnnotationPosition(qualifiers)) == NOT_FOUND) {
```

Comment:
Why not just a `boolean` method returning if annotation is present or not? To hide the low level concept of `position`?